app/riscv.ld

```diff
@@ -0,0 +1,30 @@
+/* Minimal RISC-V linker script */
+OUTPUT_ARCH("riscv")         /* Specify the target architecture */
+ENTRY(_start)                 /* Set the entry point to _start */
+
+MEMORY {
+  RAM (rwx) : ORIGIN = 0x00000000, LENGTH = 4K /* Define a 4KB RAM starting at address 0 */
+}
+
+SECTIONS {
+  .text : {
+    *(.text.init)             /* Initialization code (if any) */
+    *(.text)                  /* All code sections */
+  } > RAM
+
+  .rodata : {
+    *(.rodata)                /* Read-only data */
+  } > RAM
+
+  .data : {
```

Comment:
It's strange that you don't have explicit offset for dmem (dmem is everything that is it "text")